warehouse/warehouse.go

```diff
@@ -749,6 +759,76 @@ func setLastExec(warehouse warehouseutils.WarehouseT) {
 	lastExecMap[connectionString(warehouse)] = timeutil.Now().Unix()
 }
 
+func (wh *HandleT) handleCrashRecovery(warehouse warehouseutils.WarehouseT) error {
+	_, ok := inRecoveryMap[warehouse.Destination.ID]
+	if ok {
+		whManager, err := warehousemanager.NewWhManager(wh.destType)
+		if err != nil {
+			panic(err)
+		}
+		logger.Infof("WH: Crash recovering for %s:%s", wh.destType, warehouse.Destination.ID)
+		err = whManager.CrashRecover(warehouseutils.ConfigT{
+			DbHandle:  wh.dbHandle,
+			Warehouse: warehouse,
+		})
+		if err != nil {
+			logger.Errorf("WH: Crash recovery failed for %s:%s with error %s", wh.destType, warehouse.Destination.ID, err.Error())
+			return err
+		}
+		delete(inRecoveryMap, warehouse.Destination.ID)
+	}
+	return nil
+}
+
+func (wh *HandleT) processPendingStagingFilesViaCron() {
```

Comment:
This code is repeated again. Let's try to reuse the code instead of repeating it. Make the default behavior also push the warehouse object to channel?

Reply:
Otherwise becomes difficult to maintain.